lib/ansible/modules/cloud/online/online_manage_c14.py

```diff
@@ -0,0 +1,496 @@
+        
+        post_data = dict(
+            protocols = ['SSH'],
+            location_id = location_id,
+            key = archive_key
+        )
+        result = self.post('/%s' % self.name, post_data)
+        time.sleep(5)
+        if wait:
+            while True:
+                archive = self.get_archives(archive_uuid)
+                if archive[0]['status'] == 'active':
+                    break
+                else:
+                    time.sleep(30)
+        self.module.exit_json(result="The archive {0} has been unfreezed".format(archive_uuid))
+
+    def remove_archive(self, archive_uuid):
+        safe = self.find_safe_uuid_from_archive(archive_uuid)        
+        
+        bucket = self.get_bucket(archive_uuid)
+        if bucket is not None:
+            self.module.fail_json(msg='An open temporary space can not be removed. Freeze the archive {0} first'.format(archive_uuid))
+            
+        self.name = "storage/c14/safe/{0}/archive/{1}".format(safe,archive_uuid)
+        delete_data = dict()
+        result = self.delete('/%s' % self.name, delete_data)
+        
+        self.module.exit_json(result='The archive {0} is scheduled to be removed'.format(archive_uuid))
+
+    def get_archive_key(self, archive_uuid):
+        safe = self.find_safe_uuid_from_archive(archive_uuid)
+        self.name = "storage/c14/safe/{0}/archive/{1}/key".format(safe,archive_uuid)
+        key = self.get('/%s' % self.name)
+        self.module.exit_json(result=key.json)
+        
+    def set_archive_key(self, archive_uuid, archive_key):
+        safe = self.find_safe_uuid_from_archive(archive_uuid)
+        self.name = "storage/c14/safe/{0}/archive/{1}/key".format(safe,archive_uuid)
+        post_data = dict(
+            key = archive_key
+        )
+        result = self.post('/%s' % self.name, post_data)
+        if result.json is None:
+            self.module.exit_json(result="The archive key has been set")
+        else:        
+            self.module.exit_json(result=result.json)        
+
+    def delete_archive_key(self, archive_uuid, archive_key):
+        safe = self.find_safe_uuid_from_archive(archive_uuid)
+        self.name = "storage/c14/safe/{0}/archive/{1}/key".format(safe,archive_uuid)
+        delete_data = dict()
+        result = self.delete('/%s' % self.name, delete_data)
+        if result.json is None:
+            self.module.exit_json(result="The archive key has been deleted")
+        else:
+            self.module.exit_json(result=result.json)          
+        
+def main():
+    argument_spec = online_argument_spec()
+    argument_spec.update(
+        dict(
+            operation=dict(
+                required=True,
+                choices=['get_archives','get_safes','get_bucket','get_files','get_archive_key','set_archive_key','delete_archive_key','create_archive','upload_files','download_files','freeze_archive','unfreeze_archive','remove_archive','remove_files']
```

Comment:
Could you put all those strings in a dict as keys and for the values just use a dict. By doing so, you are sure to modify only in one place the name of functions